sdks/python/apache_beam/runners/portability/java_reference_runner_test.py

```diff
@@ -0,0 +1,163 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file is an entry point for running validatesRunner tests with the Python
+# SDK and the Java Reference Runner. Executing this file starts up an instance
+# of the Java Reference Runner's job server before executing tests and teardown
+# the job server afterwards.
+from __future__ import absolute_import
+
+import argparse
+import logging
+import sys
+import unittest
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import DebugOptions
+from apache_beam.options.pipeline_options import PortableOptions
+from apache_beam.runners.portability import portable_runner
+from apache_beam.runners.portability import portable_runner_test
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+if __name__ == '__main__':
+  # Run as
+  #
+  # python -m apache_beam.runners.portability.java_reference_runner_test \
+  #     --job_server_jar=/path/to/job_server.jar \
+  #     --environment_type=docker \
+  #     [Test.test_method, ...]
+
+  parser = argparse.ArgumentParser(add_help=True)
+  parser.add_argument('--job_server_jar',
+                      help='Job server jar to submit jobs.')
+  parser.add_argument('--environment_type', default='docker',
+                      help='Environment type. docker or process')
+  parser.add_argument('--environment_config', help='Environment config.')
+
+  known_args, args = parser.parse_known_args(sys.argv)
+  sys.argv = args
+
+  job_server_jar = known_args.job_server_jar
+  environment_type = known_args.environment_type.lower()
+  environment_config = (
+      known_args.environment_config if known_args.environment_config else None)
+
+  # This is defined here to only be run when we invoke this file explicitly.
+  class JavaReferenceRunnerTest(portable_runner_test.PortableRunnerTest):
+    _use_grpc = True
+    _use_subprocesses = True
+
+    @classmethod
+    def _subprocess_command(cls, port):
+      return [
+          'java',
+          #'-Dorg.slf4j.simpleLogger.defaultLogLevel=info'
+          '-jar', job_server_jar,
+          '--port', str(port),
+      ]
+
+    @classmethod
+    def get_runner(cls):
+      return portable_runner.PortableRunner()
+
+    def create_options(self):
+      options = super(JavaReferenceRunnerTest, self).create_options()
+      options.view_as(DebugOptions).experiments = ['beam_fn_api']
+      options._all_options['parallelism'] = 1
```

Comment:
Is this needed?

Reply:
Leaving this off seems to cause a test_sdf to fail. I haven't looked into the specifics of why yet (I'll do that when I get to implementing SDF), but the option seems necessary for now.